libsolidity/formal/CHC.cpp

```diff
@@ -1597,7 +1615,8 @@ map<unsigned, vector<unsigned>> CHC::summaryCalls(CHCSolverInterface::CexGraph c
 		q.pop();
 
 		Predicate const* nodePred = Predicate::predicate(_graph.nodes.at(node).name);
-		if (nodePred->isSummary() && (_root == root || nodePred->isInternalCall() || nodePred->isExternalCall()))
+		Predicate const* rootPred = Predicate::predicate(_graph.nodes.at(root).name);
+		if (nodePred->isSummary() && (_root == root || nodePred->isInternalCall() || nodePred->isExternalCallTrusted() || nodePred->isExternalCallUntrusted() || rootPred->isExternalCallUntrusted()))
```

Comment:
This looks ugly as one line. Not sure if my suggestion aligns to our coding style, but should try splitting into multi lines
 
```suggestion
		if (
			nodePred->isSummary() && 
			(
				_root == root || 
				nodePred->isInternalCall() || 
				nodePred->isExternalCallTrusted() || 
				nodePred->isExternalCallUntrusted() || 
				rootPred->isExternalCallUntrusted()
			)
		)
```